pandas/core/frame.py

```diff
@@ -9856,36 +9856,42 @@ def _get_data() -> DataFrame:
 
             return out
 
-        assert numeric_only is None
```

Comment:
this should still hold?

Reply:
oh i guess it should be `assert numeric_only is None or min_count != 0`?

Reply:
correct we are only using the old code path when min_count is given until _mgr.reduce can handle min_count.

why do we need to add an assert?

Reply:
not needed, just took me a minute to understand what had changed